op-challenger/game/fault/trace/vm/executor.go

```diff
@@ -33,22 +33,28 @@ type Config struct {
 	InfoFreq         uint // Frequency of progress log messages (in VM instructions)
 }
 
+type VmArgs interface {
+	Cfg() Config
+	LocalInputs() *utils.LocalGameInputs
```

Comment:
We currently do pass the local inputs down to `VmArgs`, and `VmArgs` only owns a reference to the local inputs (which can be `nil`!) The lifetime of `VmArgs` is == to the lifetime of the local inputs, assuming `VmArgs` holds the sole reference. Otherwise, the lifetime of the local inputs is > than that of `VmArgs`.

Reply:
The `Executor` is already specific to a particular set of local inputs though and has a life cycle that matches that, so it can just as easily own the local inputs.  And the `VmArgs` is actually being shared across all executors for all games (we only create one instance of it).  Since `Executor` is already setup to have a lifetime that matches the game and it is the top level of the responsibility of executing (of which creating the server args is just a small part), it make sense to keep `VmArgs` really simple and leave `Executor` holding the state. It seems conceptually a lot cleaner to me and we don't need to add a bunch of public getter that expose the config and local inputs.